department-of-veterans-affairs/vets-website
# Test version of Keyboard End to End tests and a filler test

[Cypress Commands to add additional keyboard functionality](https://github.com/department-of-veterans-affairs/va.gov-team/issues/16413)

More information about these updates:
These are additional form helper functions that utilize the `cypress-real-events` package to better emulate a user pressing a key. These allow navigating the form via keyboard (Tab), selecting options, and entering text. 

**Note:** Currently choosing an option from a select dropdown is done via typing the characters of that item. Ideally, this would use the arrow keys to go through the options and select one but with the current tools using the arrow keys within a select does not work on Mac devices. This typing method is a workaround until that is fixed in the libraries used.

**To test:**
- Checkout this branch
- Start up `vets-website`: `yarn watch`
- Start the Cypress tool: `yarn cy:open`
- Find `edu-benefits` > `tests` > `feedback-tool` > `02.feedback-tool-keyboard-navigation.cypress.spec.js` and the test should run without issue.)
